internal/pkg/cli/app_list.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+	"github.com/spf13/cobra"
+)
+
+const (
+	applicationListProjectNamePrompt = "Which project's applications would you like to list?"
+	applicationListProjectNameHelper = "A project groups all of your applications together."
+)
+
+// ListAppOpts contains the fields to collect for listing an application.
+type ListAppOpts struct {
+	ShouldOutputJSON    bool
+	ShouldShowLocalApps bool
+
+	applications  []*archer.Application
+	manager       archer.ApplicationLister
+	projectGetter archer.ProjectGetter
+	projectLister archer.ProjectLister
+
+	ws archer.Workspace
+	w  io.Writer
+
+	*GlobalOpts
+}
+
+func (opts *ListAppOpts) selectProject() (string, error) {
+	projs, err := opts.projectLister.ListProjects()
+	if err != nil {
+		return "", err
+	}
+	var projStrs []string
+	for _, projStr := range projs {
+		projStrs = append(projStrs, projStr.Name)
+	}
+	if len(projStrs) == 0 {
+		log.Infoln("There are no projects to select.")
+		return "", nil
+	}
+	proj, err := opts.prompt.SelectOne(
+		applicationListProjectNamePrompt,
+		applicationListProjectNameHelper,
+		projStrs,
+	)
+	return proj, err
+}
+
+func (opts *ListAppOpts) localAppsFilter(appNames []string) {
```

Comment:
Should we rename to `localNames` and possibly use a `map`? I got confused with all the appNames :D

```go
isLocal := make(map[string]bool)
for _, app := range opts.applications {
  if _, ok := isLocal[app.Name]; !ok {
     continue
  }
  filtered = append(filtered, app)
}
```